query-engine/schema-builder/src/output_types/query_type.rs

```diff
@@ -50,6 +52,25 @@ fn find_unique_field(ctx: &mut BuilderContext, model: &ModelRef) -> Option<Outpu
     })
 }
 
+/// Builds a "single" query arity item field (e.g. "user", "post" ...) for given model
+/// that will throw a NotFoundError if the item is not found
+fn find_unique_or_throw_field(ctx: &mut BuilderContext, model: &ModelRef) -> Option<OutputField> {
+    arguments::where_unique_argument(ctx, model).map(|arg| {
+        let field_name = format!("findUniqueOrThrow{}", model.name);
```

Comment:
What do y'all think of naming this `findUniqueUserOrThrow`? It won't matter on the client anyway (since it'll be translated as `prisma.user.findUniqueOrThrow`), I just find the construction a bit weird. It feels like we're gonna throw a user.

Reply:
Addressed in https://github.com/prisma/prisma-engines/pull/3356/commits/8481c0d4792b60cfe4b6ca7ee1e707f240adf76c